File: SmartServiceHub/SmartServiceHub-backend/src/main/java/org/example/models/Comment.java
package org.example.models;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Setter
@Getter
@Entity
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = false)
    private String text;
    @Column(nullable = false, unique = false)
    private Long post_id;
    @Column(nullable = false, unique = false)
    private Long user_id;

}

File: SmartServiceHub/SmartServiceHub-backend/src/main/java/org/example/models/Comment.kt
package org.example.models

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.example.models.dto.CommentCreateDto

@Entity
@Table(name = "comments")
class Comment(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(nullable = false)
    var content: String = "",

    @Column(nullable = false)
    var postId: Long? = null,

    @Column(nullable = false)
    var userId: Long? = null,
) {
    companion object {
        @JvmStatic
        fun formCreateDto(commentCreateDto: CommentCreateDto, user: AppUser): Comment{
            return Comment(null, commentCreateDto.getContent(), commentCreateDto.getPostId(), user.id)
        }
    }
}
